src/main/java/com/externalSorting/app/ExternalSortingRunner.java

```diff
@@ -35,21 +55,21 @@
         List<String> buffer = new ArrayList<>();
         List<File> files = new ArrayList<>();
 
-        try (BufferedReader reader = Files.newBufferedReader(Paths.get(filePath))) {
-            int count = 0;
+        try (BufferedReader reader = new MyBufferedReader(filePath)) {
+            int currentEstimatedFileSize = 0;
             for (String line = reader.readLine(); line != null; line = reader.readLine()){
                 List<String> splitLine = Arrays.asList(formatString(line));
-                count++;
-                // splitLine.removeIf(word -> word.isEmpty());
+                for (String s: splitLine) {
+                    currentEstimatedFileSize += s.length() + 1;
+                }
                 buffer.addAll(splitLine);
-                // buffer.remove("");  // remove empty strings in case of larger space between lines of text
-                if (count >= CHANGE_FILE) {
-                    count = 0;
-                    files.add(sortAndWriteSplitData(buffer, directoryPath));
+                if (currentEstimatedFileSize >= TMP_FILE_MAXSIZE) {
+                    currentEstimatedFileSize = 0;
+                    files.add(sortAndWriteData(buffer, directoryPath));
                     buffer.clear();
                 }
             }
-            files.add(sortAndWriteSplitData(buffer, directoryPath)); // write remaining data
+            files.add(sortAndWriteData(buffer, directoryPath)); // write remaining data
             buffer.clear();
 
         } catch(IOException e) {
```

Comment:
What we usually do it `catch` blocks is `e.printStackTrace()` which prints the stack trace in the console so you can follow where the error was thrown.